src/plugins/data/public/ui/ui_service.ts

```diff
@@ -24,6 +27,7 @@ export interface UiServiceStartDependencies {
 
 export class UiService implements Plugin<IUiSetup, IUiStart> {
   enhancementsConfig: ConfigSchema['enhancements'];
+  private queryStatus$ = new BehaviorSubject<QueryStatus>({ status: ResultStatus.READY });
```

Comment:
We should not do this. This would mean that we are duplicating state information in multiple places and no longer have a single source of truth. If we forget to set or clear the status correctly, the UI could show both an error and the correct result at the same time, which could lead to bugs. If we are afraid of prop drilling, lets use react context within the TopNav of the data plugin, but not have 2 places that have the same state.

Reply:
Do we still need this now?